server/src/main/kotlin/suwayomi/tachidesk/manga/impl/ChapterDownloadHelper.kt

```diff
@@ -0,0 +1,24 @@
+package suwayomi.tachidesk.manga.impl
+
+import suwayomi.tachidesk.manga.impl.download.DownloadedFilesProvider
+import suwayomi.tachidesk.manga.impl.download.FolderProvider
+import java.io.InputStream
+
+object ChapterDownloadHelper {
+    fun getImage(mangaId: Int, chapterId: Int, index: Int): Pair<InputStream, String> {
+        return provider(mangaId, chapterId).getImage(index)
+    }
+
+    fun delete(mangaId: Int, chapterId: Int): Boolean {
+        return provider(mangaId, chapterId).delete()
+    }
+
+    fun putImage(mangaId: Int, chapterId: Int, index: Int, image: InputStream): Boolean {
```

Comment:
going to change this to download since we don't need a single put image here, we can just move the entire downlod logic here. Also, building an ArchiveProvider is proving difficult since appending to zip is not working for some reason

Reply:
One more point that I was thinking about is implementing a CacheProvider, seeing if this could help us completely move out any storage logic out of ImageResponse

Reply:
Made this change in the latest commit